### PR TITLE
Modified the everflow policer script for Marvell ASIC

### PR DESCRIPTION
**Pull request by rajneeshaec**
#
Summary:
Fixes # (issue)
Modified the everflow policer script for Marvell ASIC

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X ] Test case(new/improvement)

### Back port request
- [X ] 201911
- [X ] 202012
- [X ] 202205

#### How did you do it?
Modified the everflow policer script for Marvell ASIC
Changed the CIR value according to Marvell ASIC design

#### How did you verify/test it?
test cases are passing on community testbed
everflow/test_everflow_testbed.py::TestEverflowV4IngressAclIngressMirror::test_everflow_dscp_with_policer[cli-downstream] PASSED
everflow/test_everflow_testbed.py::TestEverflowV4IngressAclIngressMirror::test_everflow_dscp_with_policer[cli-upstream] PASSED  

#### Any platform specific information?
Yes

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing ansible/roles/test/files/acstests/everflow_policer_test.py<br/>Fixing tests/everflow/everflow_test_utilities.py<br/>Fixing tests/everflow/test_everflow_testbed.py<br/><br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/everflow/everflow_test_utilities.py:37:24: E261 at least two spaces before inline comment<br/>tests/everflow/everflow_test_utilities.py:37:25: E262 inline comment should start with '# '<br/>tests/everflow/everflow_test_utilities.py:70:121: E501 line too long (142 > 120 characters)<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/everflow/everflow_test_utilities.py [hunk 1]
```diff
@@ -559,6 +559,8 @@ def remove_mirror_config(duthost, session_name, config_method=CONFIG_MODE_CLI):
         duthost.command(command)
 
     def apply_policer_config(self, duthost, policer_name, config_method, rate_limit=100):
+        if duthost.facts["asic_type"] == "marvell":
+            rate_limit = 125
```
[the remark this answers is not in the document]
Comment (Blueve) [reply tests/everflow/everflow_test_utilities.py:563]: @rajneeshaec

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing ansible/roles/test/files/acstests/everflow_policer_test.py<br/>Fixing tests/everflow/everflow_test_utilities.py<br/>Fixing tests/everflow/test_everflow_testbed.py<br/><br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/everflow/everflow_test_utilities.py:37:24: E261 at least two spaces before inline comment<br/>tests/everflow/everflow_test_utilities.py:37:25: E262 inline comment should start with '# '<br/>tests/everflow/everflow_test_utilities.py:70:121: E501 line too long (142 > 120 characters)<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/everflow/everflow_test_utilities.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajneeshaec) [reply tests/everflow/everflow_test_utilities.py]: Hi @bingwang-ms,
   In Marvell ASIC, in the packet mode CIR need to be configured as (required_CIR_in_PPS * 8) / 1000
   if rate_limit is 100, then then CIR  = (100*8)/1000 < 1 is resulting to 0kbps.
   so minimum rate_limit value to be configured for the policer to work is >= 125
   when rate_limit is 125 then CIR will be (125*8)/1000 = 1.2, rounded to 1kbps, hence the policer works
   I have modified the script according to the review comment. Please review it

[the remark this answers is not in the document]
Comment (radha-danda) [reply tests/everflow/everflow_test_utilities.py]: @rajneeshaec, addressed your comments. @Blueve, can you pls review the PR?

**Review by Blueve — APPROVED** [review]

**Review by bingwang-ms — APPROVED** [review]

File: tests/everflow/everflow_test_utilities.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply tests/everflow/everflow_test_utilities.py]: > Hi @bingwang-ms, In Marvell ASIC, in the packet mode CIR need to be configured as (required_CIR_in_PPS * 8) / 1000 if rate_limit is 100, then then CIR = (100_8)/1000 < 1 is resulting to 0kbps. so minimum rate_limit value to be configured for the policer to work is >= 125 when rate_limit is 125 then CIR will be (125_8)/1000 = 1.2, rounded to 1kbps, hence the policer works I have modified the script according to the review comment. Please review it
> 
> Thanks Rajneesh

Thanks for the clarification. LGTM.

**Pull request merged**